src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -37,6 +46,270 @@
+									  char *newConstraintName);
+
+
+PG_FUNCTION_INFO_V1(fix_pre_citus10_partitioned_table_constraint_names);
+PG_FUNCTION_INFO_V1(worker_fix_pre_citus10_partitioned_table_constraint_names);
+
+
+/*
+ * fix_pre_citus10_partitioned_table_constraint_names fixes the constraint names of
+ * partitioned table shards on workers.
+ *
+ * Constraint names for partitioned table shards should have shardId suffixes if and only
+ * if they are unique or foreign key constraints. We mistakenly appended shardIds to
+ * constraint names on ALTER TABLE dist_part_table ADD CONSTRAINT .. queries prior to
+ * Citus 10. fix_pre_citus10_partitioned_table_constraint_names determines if this is the
+ * case, and renames constraints back to their original names on shards.
+ */
+Datum
+fix_pre_citus10_partitioned_table_constraint_names(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	EnsureCoordinator();
+
+	if (!PartitionedTable(relationId))
+	{
+		ereport(ERROR, (errmsg("could not fix partition constraints: "
+							   "relation does not exist or is not partitioned")));
+	}
+	if (!IsCitusTable(relationId))
+	{
+		ereport(ERROR, (errmsg("fix_pre_citus10_partitioned_table_constraint_names can "
+							   "only be called for distributed partitioned tables")));
+	}
+
+	List *taskList = CreateFixPartitionConstraintsTaskList(relationId);
+	bool localExecutionSupported = true;
+	ExecuteUtilityTaskList(taskList, localExecutionSupported);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * worker_fix_pre_citus10_partitioned_table_constraint_names fixes the constraint names on a worker given a shell
+ * table name and shard id.
+ */
+Datum
+worker_fix_pre_citus10_partitioned_table_constraint_names(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	int64 shardId = PG_GETARG_INT32(1);
+	text *constraintNameText = PG_GETARG_TEXT_P(2);
+
+	if (!PartitionedTable(relationId))
+	{
+		ereport(ERROR, (errmsg("could not fix partition constraints: "
+							   "relation does not exist or is not partitioned")));
+	}
+
+	char *constraintName = text_to_cstring(constraintNameText);
+	char *shardIdAppendedConstraintName = pstrdup(constraintName);
+	AppendShardIdToName(&shardIdAppendedConstraintName, shardId);
+
+	/* if shardId was appended to the constraint name, rename back to original */
+	if (RelationHasConstraint(relationId, shardIdAppendedConstraintName))
+	{
+		char *renameConstraintDDLCommand =
+			RenameConstraintCommand(relationId, shardIdAppendedConstraintName,
+									constraintName);
+		ExecuteAndLogUtilityCommand(renameConstraintDDLCommand);
+	}
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * CreateFixPartitionConstraintsTaskList goes over all the partitions of a distributed
+ * partitioned table, and creates the list of tasks to execute
+ * worker_fix_pre_citus10_partitioned_table_constraint_names UDF on worker nodes.
+ */
+static List *
+CreateFixPartitionConstraintsTaskList(Oid relationId)
+{
+	/* resulting task list */
```

Comment:
This comment seems unnecessary, if `taskList` is not descriptive enough, you could rename the variable instead of adding such a comment.

Reply:
Minor: I would also remove the other comments in this method as I don't think they add a value

Reply:
These comments are left over from when it was too complex that I felt the need to explain everything. Now it makes sense to remove them. Thanks for pointing it out